lib/util/package-rules/sourceurls.ts

```diff
@@ -10,13 +11,12 @@ export class SourceUrlsMatcher extends Matcher {
     if (is.undefined(matchSourceUrls)) {
       return null;
     }
-    if (is.undefined(sourceUrl)) {
+    if (!sourceUrl) {
       return false;
     }
 
-    const upperCaseSourceUrl = sourceUrl?.toUpperCase();
-    return matchSourceUrls.some(
-      (url) => upperCaseSourceUrl === url.toUpperCase(),
-    );
+    const upperCaseSourceUrl = sourceUrl.toUpperCase();
+    const upperCaseMatches = matchSourceUrls.map((url) => url.toUpperCase());
+    return matchRegexOrGlobList(upperCaseSourceUrl, upperCaseMatches);
```

Comment:
i thought glob is already case-insensitive, so no need to convert to upper case
```suggestion
    return matchRegexOrGlobList(sourceUrl, matchSourceUrls);
```